cirq-core/cirq/sim/density_matrix_simulator_test.py

```diff
@@ -1527,3 +1529,27 @@ def test_density_matrices_same_with_or_without_split_untangled_states():
     sim = cirq.DensityMatrixSimulator(split_untangled_states=True)
     result2 = sim.simulate(circuit).final_density_matrix
     assert np.allclose(result1, result2)
+
+
+def test_large_untangled_okay():
+    circuit = cirq.Circuit()
+    for i in range(59):
+        for _ in range(9):
+            circuit.append(cirq.X(cirq.LineQubit(i)))
+        circuit.append(cirq.measure(cirq.LineQubit(i)))
+
+    # Validate this can't be allocated with entangled state
+    if platform.system() != "Windows":
```

Comment:
Why does this behave differently on Windows? We should aim for consistent behavior across all platforms.

Reply:
I remember it throws a different error prior to actually trying to allocate, but can't recall the specific error.

Reply:
Oh here it is: https://github.com/quantumlib/Cirq/pull/4198/checks?check_run_id=2813139769

It looks like the computation 2^59 in numpy is overflowing on Windows. I can change the test to just catch any exception, and that would fix the short-term. But we should probably consider setting numpy to fail on overflow errors too. Do you know is there a config or some way to do this globally? (or even locally? I'm not finding anything)

Reply:
I think you need to choose a different `dtype` in the test: `dtype=int` is likely system-dependent and resulting in 32-bit ints on the Windows machine.

Reply:
It's not the test code, it's code in `qis.states::quantum_state` that's doing it. There's quite a few places (search for np.prod in https://github.com/quantumlib/Cirq/blob/master/cirq-core/cirq/qis/states.py), and probably elsewhere too.

Should we change all these to use Python numerics so they don't overflow?

Reply:
You're right - I guess we made a baseline assumption that 32 (or 64) qubits would be enough for anyone. For the sake of keeping this PR focused, let's keep the test as-is for now but leave a TODO (plus a Github issue) to update the numerics.

Reply:
#4363 should unblock this (assuming the statements made here: https://numpy.org/doc/stable/reference/arrays.scalars.html#numpy.int_) hold true for windows as well.